### PR TITLE
Add initial Travis CI config.

### PR DESCRIPTION
**Pull request by gavanderhoorn**
As per subject.

This is a test.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+cache:
+  directories:
+    - $HOME/.ccache
+
+git:
+  quiet: true
+
+env:
+  global:
+    - CATKIN_LINT=true
```
Comment (gavanderhoorn) [inline .travis.yml:14]: Not sure we want to enable this immediately.

@cardboardcode: will the packages as they are pass a `catkin_lint` check?

Comment (cardboardcode) [reply .travis.yml:14]: @gavanderhoorn Thank you for the quick response.

The packages will **fail** with `catkin_lint` check.

**Reason**:  There are dependencies on external packages beyond **rvip**. eg.[**darknet_ros**](https://github.com/leggedrobotics/darknet_ros)

**Follow-up**: Will remove those dependencies shortly and update again in this thread.

Comment (gavanderhoorn) [reply .travis.yml:14]: > The packages will **fail** with `catkin_lint` check.

Are you sure? Afaik `catkin_lint` does not check for presence of dependencies. It just checks `package.xml` and `CMakeLists.txt` for problems.

> **Follow-up**: Will remove those dependencies shortly and update again in this thread.

If those dependencies would need to be removed for CI to pass, we need to fix CI (within reason of course), not the dependencies.

Comment (cardboardcode) [reply .travis.yml:14]: HI @gavanderhoorn,

The codebase should now pass the `catkin_lint` check with [**the recent commit**](https://github.com/ros-industrial/rvip/pull/1/commits/2e747a5f3b0c4059d6dafcc76f5fc073e2fe699d).

Please proceed with enabling.

Comment (gavanderhoorn) [reply .travis.yml:14]: If you could review this PR (ie: *Accept* it), I can merge the PR.

**Review by Levi-Armstrong — APPROVED** [review]

**Comment by gavanderhoorn** [comment]
Thanks @Levi-Armstrong.

@cardboardcode: if you could rebase your PR onto `master`, it should start automatically running CI for your PR as well.

**Comment by cardboardcode** [comment]
Thanks @gavanderhoorn 

Pull Request #1 has been rebased onto `master`. `travis.yml` has been added. 

**Edit**: The build failed. I have attempted this integration before but still unable to determine the specifics of the build error. Due to this, I had resorted to using **gitlab-ci** with my own catkin build process.

Is there a way to generate a more detailed build error output by **industrial-ci**? Please advise.

> I have tried modifying **industrial-ci** to run `catkin build/make` with `--verbose` flag.

**Comment by gavanderhoorn** [comment]
I've added a comment over at #2. Let's continue to diagnose why CI fails there.

**Pull request merged**